lib/BuildSystem/BuildSystemFrontend.cpp

```diff
@@ -232,12 +232,21 @@ class BuildSystemFrontendExecutionQueueDelegate
 };
 
 struct BuildSystemFrontendDelegateImpl {
+
+  /// The status of delegate.
+  enum class Status {
+    Uninitialized = 0,
```

Comment:
Actually, we should also have "FailedToInitialize" or something, so that we can report an error if initialization fails but the client tries to build anyway.